src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -81,7 +83,30 @@ object Ledger extends Logger {
 
   private def validateBlockBeforeExecution(block: Block): Option[String] = None
 
-  private def validateBlockAfterExecution(block: Block, receipts: Seq[Receipt], worldStateProxy: InMemoryWorldStateProxy): Option[String] = None //TODO
+  /**
+    * This function validates that the various results from execution are consistent with the block. This includes:
+    *   - Validating the resulting stateRootHash
+    *   - Doing BlockValidator.validateBlockReceipts validations involving the receipts
+    *   - Validating the resulting gas used
+    *
+    * @param block to validate
+    * @param stateRootHash from the resulting state trie after executing the txs from the block
+    * @param receipts associated with the execution of each of the tx from the block
+    * @param gasUsed, accumulated gas used for the execution of the txs from the block
+    * @return None if valid else a message with what went wrong
+    */
+  private[ledger] def validateBlockAfterExecution(block: Block, stateRootHash: ByteString,
+                                                  receipts: Seq[Receipt], gasUsed: BigInt): Option[String] = {
+    lazy val blockAndReceiptsValidation = BlockValidator.validateBlockAndReceipts(block, receipts)
+    if(block.header.gasUsed != gasUsed)
+      Some(s"Block has invalid gas used: ${block.header.gasUsed} != $gasUsed")
+    else if(block.header.stateRoot != stateRootHash)
+      Some(s"Block has invalid state root hash: ${block.header.stateRoot} != $stateRootHash")
+    else if(blockAndReceiptsValidation.isLeft)
+      Some(blockAndReceiptsValidation.left.get.toString)
+    else
+      None
```

Comment:
So that just gives us the possibility accumulate checks. All the validations we currently give up on first failure, so I wouldn't do anything else here. Shouldn't be a problem to change that in the future.